assets/fluid_simulation/shaders/sunrays_mask.hlsl

```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 Pavel Dobryakov
+// Copyright 2022 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "config.hlsli"
+
+[numthreads(1, 1, 1)] void csmain(uint2 tid
+                                  : SV_DispatchThreadID) {
+    Coord  coord = BaseVS(tid, Params.normalizationScale, Params.texelSize);
+    float4 c     = UTexture.SampleLevel(Sampler, coord.vUv, 0);
+    float  br    = max(c.r, max(c.g, c.b));
+    c.a          = 1.0 - min(max(br * 20.0, 0.0), 0.8);
+    Output[coord.xy] = c;
```

Comment:
Is it portable to consider the input as float4 while it's a single  texture?

Reply:
You mean `c`?  It's an RGBA value, right?  Not sure what you mean by this.